Source/Particles/Deposition/VarianceAccumulationBuffer.cpp

```diff
@@ -0,0 +1,212 @@
+/* Copyright 2025 The WarpX Community
+ *
+ * This file is part of WarpX.
+ *
+ * Authors: S. Eric Clark (Helion Energy)
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "Particles/Deposition/VarianceAccumulationBuffer.H"
+#include "Particles/Deposition/TemperatureDeposition.H"
+#include "Parallelization/WarpXSumGuardCells.H"
+#include "Fields.H"
+
+#include <ablastr/utils/Communication.H>
+
+#include <AMReX.H>
+#include <AMReX_REAL.H>
+
+using namespace amrex::literals;
+using namespace warpx::particles::deposition;
+
+VarianceAccumulationBuffer::VarianceAccumulationBuffer (ablastr::fields::MultiLevelVectorField const& T_vf, std::string const& species_name )
+{
+    using ablastr::fields::Direction;
+    auto & warpx = WarpX::GetInstance();
+
+    const int ncomps = 1;
+
+    for (int lev = 0; lev <= warpx.finestLevel(); ++lev) {
+        amrex::BoxArray const& bax = T_vf[lev][Direction{0}]->boxArray();
+        amrex::BoxArray const& bay = T_vf[lev][Direction{1}]->boxArray();
+        amrex::BoxArray const& baz = T_vf[lev][Direction{2}]->boxArray();
+
+        amrex::DistributionMapping const& dmx = T_vf[lev][Direction{0}]->DistributionMap();
+        amrex::DistributionMapping const& dmy = T_vf[lev][Direction{1}]->DistributionMap();
+        amrex::DistributionMapping const& dmz = T_vf[lev][Direction{2}]->DistributionMap();
+
+        amrex::IntVect const& ngx = T_vf[lev][Direction{0}]->nGrowVect();
+        amrex::IntVect const& ngy = T_vf[lev][Direction{1}]->nGrowVect();
+        amrex::IntVect const& ngz = T_vf[lev][Direction{2}]->nGrowVect();
+
+        warpx.m_fields.alloc_init("w_" + species_name, Direction{0}, lev, bax, dmx, ncomps, ngx, 0.0_rt);
+        warpx.m_fields.alloc_init("w_" + species_name, Direction{1}, lev, bay, dmy, ncomps, ngy, 0.0_rt);
+        warpx.m_fields.alloc_init("w_" + species_name, Direction{2}, lev, baz, dmz, ncomps, ngz, 0.0_rt);
+
+        warpx.m_fields.alloc_init("w2_" + species_name, Direction{0}, lev, bax, dmx, ncomps, ngx, 0.0_rt);
+        warpx.m_fields.alloc_init("w2_" + species_name, Direction{1}, lev, bay, dmy, ncomps, ngy, 0.0_rt);
+        warpx.m_fields.alloc_init("w2_" + species_name, Direction{2}, lev, baz, dmz, ncomps, ngz, 0.0_rt);
+
+        warpx.m_fields.alloc_init("vbar_" + species_name, Direction{0}, lev, bax, dmx, ncomps, ngx, 0.0_rt);
+        warpx.m_fields.alloc_init("vbar_" + species_name, Direction{1}, lev, bay, dmy, ncomps, ngy, 0.0_rt);
+        warpx.m_fields.alloc_init("vbar_" + species_name, Direction{2}, lev, baz, dmz, ncomps, ngz, 0.0_rt);
+    }
+
+    // Grab references to these arrays to store in this data structure
+    this->w = warpx.m_fields.get_mr_levels_alldirs("w_" + species_name, warpx.finestLevel());
```

Comment:
It would be better to fetch the MFs from the registry each time they are needed rather than saving references. The main issues is that during a load balance, these saved references would become invalid. Here, you would want to save only the name of the MF.